pulsar-broker/src/main/java/org/apache/pulsar/broker/service/TopicListService.java

```diff
@@ -59,9 +65,12 @@ public List<String> getMatchingTopics() {
             return matchingTopics;
         }
 
+        /***
+         * @param topicName topic name which contains partition suffix.
+         */
         @Override
         public void accept(String topicName, NotificationType notificationType) {
-            if (topicsPattern.matcher(topicName).matches()) {
+            if (topicsPattern.matcher(TopicName.get(topicName).getPartitionedTopicName()).matches()) {
```

Comment:
The problem is the topic name with a partition suffix can't match the subscription regexp topic name, right?